src/search/landmarks/landmark_heuristic.cc

```diff
@@ -150,34 +152,42 @@ void LandmarkHeuristic::generate_preferred_operators(
 }
 
 int LandmarkHeuristic::compute_heuristic(const State &ancestor_state) {
-    State state = convert_ancestor_state(ancestor_state);
-    lm_status_manager->update_lm_status(ancestor_state);
-    int h = get_heuristic_value(state);
-
+    int h = get_heuristic_value(ancestor_state);
     if (use_preferred_operators) {
-        BitsetView reached_lms =
-            lm_status_manager->get_reached_landmarks(ancestor_state);
-        generate_preferred_operators(state, reached_lms);
+        ConstBitsetView past = lm_status_manager->get_past_landmarks_const(ancestor_state);
+        State state = convert_ancestor_state(ancestor_state);
+        generate_preferred_operators(state, past);
     }
-
     return h;
 }
 
 void LandmarkHeuristic::notify_initial_state(const State &initial_state) {
-    lm_status_manager->process_initial_state(initial_state, log);
+    lm_status_manager->progress_initial_state(initial_state);
 }
 
 void LandmarkHeuristic::notify_state_transition(
     const State &parent_state, OperatorID op_id, const State &state) {
-    lm_status_manager->process_state_transition(parent_state, op_id, state);
+    lm_status_manager->progress(parent_state, op_id, state);
     if (cache_evaluator_values) {
-        /* TODO:  It may be more efficient to check that the reached landmark
+        /* TODO:  It may be more efficient to check that the past landmark
             set has actually changed and only then mark the h value as dirty. */
         heuristic_cache[state].dirty = true;
     }
 }
 
 void LandmarkHeuristic::add_options_to_feature(plugins::Feature &feature) {
+    feature.document_synopsis(
+        "Landmark progression is implemented according to the following paper:"
+        + utils::format_conference_reference(
+            {"Clemens Büchner", "Thomas Keller", "Salomé Eriksson", "Malte Helmert"},
+            "Landmarks Progression in Heuristic Search",
+            "https://ai.dmi.unibas.ch/papers/buechner-et-al-icaps2023.pdf",
+            "Proceedings of the Thirty-Third International Conference on "
+            "Automated Planning and Scheduling (ICAPS 2023)",
+            "TODO", // TODO: Pages missing.
+            "AAAI Press",
+            "2023"));
+
```

Comment:
> This TODO still needs to be addressed. Do we follow the paper 100%, or is there a difference worth mentioning?

Reply:
I'm not aware of any differences. Also, I've added the pages which are now published.